asteroid/engine/system.py

```diff
@@ -33,21 +35,27 @@ class System(pl.LightningModule):
     """
 
     def __init__(
-        self,
-        model,
-        optimizer,
-        loss_func,
-        train_loader,
-        val_loader=None,
-        scheduler=None,
-        config=None,
+            self,
+            model,
+            optimizer,
+            loss_func,
+            train_loader,
+            sample_rate,
+            val_loader=None,
+            test_loader=None,
+            metrics_list=[],
```

Comment:
I'm not sure that we want to have this here. I don't like mixing the test loader with all the rest. 

Maybe we should have a separate method `prepare_for_test` or only use the `trainer.test(system, test_data)`?
 
I'd like @jonashaag and @mhu-coder point of view here.

Reply:
Good question... how would it be done with PL? From a look at their docs it seems there is only `trainer.test()` which uses the `Metric`s that you added to your module? (Or whatever you do for logging.) With no results export (to dict, CSV, ...)?

Reply:
The test loader can be passed at test time only, all the loaders can be outside of the `LightningModule` for that matter.